### PR TITLE
Add C# example to Playing videos

### PR DESCRIPTION
**Pull request by ShawnHardern**
Add C# examples to Playing videos

### CODE REVIEW

File: tutorials/animation/playing_videos.rst [hunk 1]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+ .. code-tab:: gdscript
 
-   extends Control
+    extends Control
 
-   func _on_color_picker_button_color_changed(color):
-       # Update the "chroma_key_color" shader parameter of the VideoStreamPlayer's material
-       $VideoStreamPlayer.material.set("shader_parameter/chroma_key_color", color)
+    func _on_color_picker_button_color_changed(color):
+        # Update the "chroma_key_color" shader parameter of the VideoStreamPlayer's material
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:320]:
```suggestion
        # Update the "chroma_key_color" shader parameter of the VideoStreamPlayer's material.
```
While we're already here, per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).

File: tutorials/animation/playing_videos.rst [hunk 2]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+        $VideoStreamPlayer.material.set("shader_parameter/chroma_key_color", color)
 
-   func _on_h_slider_value_changed(value):
-       # Update the "pickup_range" shader parameter of the VideoStreamPlayer's material
-       $VideoStreamPlayer.material.set("shader_parameter/pickup_range", value)
+    func _on_h_slider_value_changed(value):
+        # Update the "pickup_range" shader parameter of the VideoStreamPlayer's material
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:324]:
```suggestion
        # Update the "pickup_range" shader parameter of the VideoStreamPlayer's material.
```

File: tutorials/animation/playing_videos.rst [hunk 3]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+        $VideoStreamPlayer.material.set("shader_parameter/pickup_range", value)
 
-   func _on_h_slider_2_value_changed(value):
-       # Update the "fade_amount" shader parameter of the VideoStreamPlayer's material
-       $VideoStreamPlayer.material.set("shader_parameter/fade_amount", value)
+    func _on_h_slider_2_value_changed(value):
+        # Update the "fade_amount" shader parameter of the VideoStreamPlayer's material
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:328]:
```suggestion
        # Update the "fade_amount" shader parameter of the VideoStreamPlayer's material.
```

File: tutorials/animation/playing_videos.rst [hunk 4]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+        $VideoStreamPlayer.material.set("shader_parameter/fade_amount", value)
 
    func _on_video_stream_player_finished():
-       # Restart the video playback when it's finished
-       $VideoStreamPlayer.play()
+        # Restart the video playback when it's finished
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:332]:
```suggestion
        # Restart the video playback when it's finished.
```

File: tutorials/animation/playing_videos.rst [hunk 5]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+            _videoStreamPlayer.Finished += OnVideoStreamPlayerFinished;
+
+            GetNode<ColorPickerButton>("ColorPickerButton").ColorChanged += OnColorPickerButtonColorChanged;
+            GetNode<HSlider>("HSlider").ValueChanged += OnHSliderValueChanged;
+            GetNode<HSlider>("HSlider2").ValueChanged += OnHSlider2ValueChanged;
+        }
+
+        void OnColorPickerButtonColorChanged(Color color)
+        {
+            // Update the "chroma_key_color" shader parameter of the VideoStreamPlayer's material
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:353]:
```suggestion
            // Update the "chroma_key_color" shader parameter of the VideoStreamPlayer's material.
```

File: tutorials/animation/playing_videos.rst [hunk 6]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+            _videoStreamPlayer.Material.Set("shader_parameter/chroma_key_color", color);
+        }
+
+        private void OnHSliderValueChanged(double value)
+        {
+            // Update the "pickup_range" shader parameter of the VideoStreamPlayer's material
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:359]:
```suggestion
            // Update the "pickup_range" shader parameter of the VideoStreamPlayer's material.
```

File: tutorials/animation/playing_videos.rst [hunk 7]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+            _videoStreamPlayer.Material.Set("shader_parameter/pickup_range", value);
+        }
+
+        private void OnHSlider2ValueChanged(double value)
+        {
+            // Update the "fade_amount" shader parameter of the VideoStreamPlayer's material
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:365]:
```suggestion
            // Update the "fade_amount" shader parameter of the VideoStreamPlayer's material.
```

File: tutorials/animation/playing_videos.rst [hunk 8]
```diff
@@ -311,25 +311,67 @@ UI Controls
...
+            _videoStreamPlayer.Material.Set("shader_parameter/fade_amount", value);
+        }
+
+        private void OnVideoStreamPlayerFinished()
+        {
+            // Restart the video playback when it's finished
```
Comment (AThousandShips) [inline tutorials/animation/playing_videos.rst:371]:
```suggestion
            // Restart the video playback when it's finished.
```

File: tutorials/animation/playing_videos.rst (hunk 1 above)
Comment (ShawnHardern) [reply tutorials/animation/playing_videos.rst:320]: Great spot! 😉 @AThousandShips

**Review by raulsntos — COMMENTED** [review]
I also noticed the code block in the _Writing the Custom Shader_ section uses `gd` but it looks like we use `glsl` for shaders everywhere else. We may want to change it while we're updating this page.

**Comment by ShawnHardern** [comment]
> I also noticed the code block in the _Writing the Custom Shader_ section uses `gd` but it looks like we use `glsl` for shaders everywhere else. We may want to change it while we're updating this page.

Great catch!  ill update that too 😎

File: tutorials/animation/playing_videos.rst [hunk 9]
```diff
@@ -311,25 +311,64 @@ UI Controls
...
-       # Restart the video playback when it's finished
-       $VideoStreamPlayer.play()
+        # Restart the video playback when it's finished.
+        $VideoStreamPlayer.play()
+
+ .. code-tab:: csharp
+
+    using Godot;
+
+    public partial class MyClass : Control
```
Comment (raulsntos) [inline tutorials/animation/playing_videos.rst:339]: My bad, I forgot to fix the name in the previous review.

```suggestion
    public partial class MyControl : Control
```

Comment (ShawnHardern) [reply tutorials/animation/playing_videos.rst:339]: Ah I missed this too ! 😂

**Review by raulsntos — APPROVED** [review]
Looks good to me, thanks!

**Review by AThousandShips — APPROVED** [review]

**Comment by AThousandShips** [comment]
Thanks!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.3 in https://github.com/godotengine/godot-docs/pull/10038.

**Pull request merged**